### ISSUE TITLE
docker build -t shoop/dev .   --->   non-zero code: 1  ??

### ISSUE THREAD

**Issue by jujes** [shuup/shuup#482]
**I get this issue, someone know why?**

```
$ uname -r
  3.13.0-86-generic
$ docker --version
Docker version 1.11.1, build 5604cbe
$ git clone git clone https://github.com/suutari-ai/shoop.git
$ cd shoop && docker build -t shoop/dev .

```
`The command '/bin/sh -c /var/www/shoop/venv/bin/pip install /var/www/shoop/working_copy' returned a non-zero code: 1
`

**Comment by suutari-ai** [comment]
Hi, @jujes, and thanks for reporting this.

The Docker builds aren't currently used by us for anything and therefore we're not actively checking that they work. However, we would like make Shoop easy to install and it would be nice to have a working Docker image too, as on option for trying out Shoop.

I checked more details about the error by triggering a build from fresh master at Docker Hub and got a more detailed error log: https://hub.docker.com/r/shoopio/shoop/builds/b6kr7zen9wmcpnzu4pujy2a/

The log is quite unreadable, but I managed to find a fatal error from it:
```
bower ESUDO Cannot be run with sudo

Additional error details:
Since bower is a user command, there is no need to execute it with superuser permissions.
If you're having permission errors when using bower without sudo, please spend a few minutes learning more about how your system should work and make any necessary repairs.
```

So bower cannot be ran as root, as our Dockerfile currently does. (Found similar issues on other projects too, like this one: https://github.com/GeoNode/geonode/pull/1070)

So, I see two possible ways to fix this:
 * (A) Add a separate `pip wheel /var/www/shoop/working_copy` to the Dockefile and run it as non-root user. Then do the `pip install` part using the generated wheel file, i.e. something like `pip install dist/shoop*.wheel`
 * (B) Add the `--allow-root` option to our Bower invocations.

I think A is better and shouldn't be hard to implement. Want to give it a try?

**Comment by jujes** [comment]
Hi @suutari-ai 

I dont' know how separate pip whell in the Dockerfile ..

now this is my output:

```
  Running setup.py install for shuup: started
    Running setup.py install for shuup: finished with status 'error'
    Complete output from command /var/www/shuup/venv/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-latoanes-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\
n', '\n'), __file__, 'exec'))" install --record /tmp/pip-cdgjmfy3-record/install-record.txt --single-version-externally-managed --compile --install-headers /var/www/shuup/venv/include/site/python3.4/shuup:
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.4/logging/__init__.py", line 980, in emit
    stream.write(msg)
UnicodeEncodeError: 'ascii' codec can't encode characters in position 1561-1563: ordinal not in range(128)
Call stack:
  File "/var/www/shuup/venv/bin/pip", line 11, in <module>
    sys.exit(main())
  File "/var/www/shuup/venv/lib/python3.4/site-packages/pip/__init__.py", line 221, in main
    return command.main(cmd_args)
  File "/var/www/shuup/venv/lib/python3.4/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/var/www/shuup/venv/lib/python3.4/site-packages/pip/commands/install.py", line 317, in run
    prefix=options.prefix_path,
  File "/var/www/shuup/venv/lib/python3.4/site-packages/pip/req/req_set.py", line 742, in install
    **kwargs
  File "/var/www/shuup/venv/lib/python3.4/site-packages/pip/req/req_install.py", line 880, in install
    spinner=spinner,
  File "/var/www/shuup/venv/lib/python3.4/site-packages/pip/utils/__init__.py", line 714, in call_subprocess
    '\n----------------------------------------'
Message: 'running install\nrunning build\nrunning build_production_resources\n*** (1/5) Building: ./shuup/xtheme\n\n> shuup-xtheme@1.0.0 build
```

**Comment by suutari-ai** [comment]
Fixed by PR #873.
